dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -183,42 +169,68 @@ public Response organizationsGet(SecurityContext securityContext) {
-                String textTools = Resources.toString(urlTools, StandardCharsets.UTF_8);
-
-                // Get mapping for workflows index
-                URL urlWorkflows = Resources.getResource("queries/mapping_workflow.json");
-                String textWorkflows = Resources.toString(urlWorkflows, StandardCharsets.UTF_8);
-
-                // Create indices
-                CreateIndexRequest toolsRequest = new CreateIndexRequest(TOOLS_INDEX);
-                toolsRequest.source(textTools, XContentType.JSON);
-                CreateIndexRequest workflowsRequest = new CreateIndexRequest(WORKFLOWS_INDEX);
-                workflowsRequest.source(textWorkflows, XContentType.JSON);
-                client.indices().create(toolsRequest, RequestOptions.DEFAULT);
-                client.indices().create(workflowsRequest, RequestOptions.DEFAULT);
-
-                // Populate index
-                if (!published.isEmpty()) {
-                    publicStateManager.bulkUpsert(published);
-                }
-            } catch (IOException e) {
-                LOG.error("Could not create elastic search index", e);
-                throw new CustomWebApplicationException("Search indexing failed", HttpStatus.SC_INTERNAL_SERVER_ERROR);
-            }
-            return Response.ok().entity(published.size()).build();
+            clearIndex();
+            int totalProcessed = 0;
+            LOG.info("Starting GA4GH batch processing");
+            totalProcessed += processDAO(toolDAO);
+            LOG.info("Processed {} tools", totalProcessed);
+            totalProcessed += processDAO(workflowDAO);
+            LOG.info("Processed {} tools and workflows", totalProcessed);
+            totalProcessed += processDAO(appToolDAO);
+            LOG.info("Processed {} tools, workflows, and apptools", totalProcessed);
+            return Response.ok().entity(totalProcessed).build();
         }
         return Response.ok().entity(0).build();
     }
 
+    private int processDAO(EntryDAO entryDAO) {
+        int processed = 0;
+        List<? extends Entry> published;
+        do {
+            published = entryDAO.findAllPublished(processed, ES_BATCH_INSERT_SIZE, null, "gitUrl", "asc");
+            processed += published.size();
+            processBatch(published);
+        } while (published.size() == ES_BATCH_INSERT_SIZE);
+        return processed;
+    }
+
+    private void clearIndex() {
+        try {
+            // FYI. it is real tempting to use a try ... catch with resources to close this client, but it actually permanently messes up the client!
+            RestHighLevelClient client = ElasticSearchHelper.restHighLevelClient();
+            // Delete previous indices
+            deleteIndex(client, TOOLS_INDEX);
+            deleteIndex(client, WORKFLOWS_INDEX);
+
+            // Get mapping for tools index
+            URL urlTools = Resources.getResource("queries/mapping_tool.json");
```

Comment:
This code already existed, so no big deal if you don't, but it would be clearer and more DRY if the index creation process was encapsulated in a function, and that function was called from here, ala:
```
    createIndex(client, "queries/mapping_tool.json", TOOLS_INDEX);
    createIndex(client, "queries/mapping_workflow.json", WORKFLOWS_INDEX);
```